### PR TITLE
Rescoring sprint filters

### PR DESCRIPTION
**Pull request by 8R0WNI3**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

File: src/components/dependencies/BDBARescoring.js [hunk 1]
```diff
@@ -568,6 +565,44 @@ VulnerabilityExtraInfo.propTypes = {
 }
 
 
+const AppliedRulesExtraInfo = ({
+  matchingRules,
+}) => {
+  if (matchingRules.every((rule) => ['original-severity', 'custom-rescoring'].includes(rule))) {
```
Comment (zkdev) [inline src/components/dependencies/BDBARescoring.js:662]: Please introduce a enum/frozen-object for "meta rules"

File: src/components/dependencies/BDBARescoring.js [hunk 2]
```diff
@@ -375,7 +380,85 @@ RescoringHeader.propTypes = {
...
+  selectedSprints,
+  setSelectedSprints,
+  sprints,
+}) => {
+  const searchParamContext = React.useContext(SearchParamContext)
+  const theme = useTheme()
+
+  if (sprints.length === 0) return null
+
+  // if there are any sprints selected which don't have any associated findings, remove the selection
```
Comment (zkdev) [inline src/components/dependencies/BDBARescoring.js:393]: This comment confuses me.
Code checks whether all selected sprints are valid.

Why is that required at all?
Shouldn't we only offer selection of "valid" sprints?

File: src/components/dependencies/BDBARescoring.js [hunk 3]
```diff
@@ -1829,6 +1968,68 @@ Rescore.propTypes = {
...
+  const now = new Date(new Date().setHours(0, 0, 0, 0))
+
+  const findingIsResolved = () => {
+    if (rescoring.applicable_rescorings.length === 0) return false
+    const rescoringsOrderedBySpecificity = orderRescoringsBySpecificity(rescoring.applicable_rescorings)
+    return rescoringsOrderedBySpecificity[0].data.severity === SEVERITIES.NONE
+  }
+
+  if (findingIsResolved())
+    return 'Resolved'
```
Comment (zkdev) [inline src/components/dependencies/BDBARescoring.js:1981]: Also introduce an enum/frozen-object for sprint "meta names"

Comment (zkdev) [reply src/components/dependencies/BDBARescoring.js:1981]: This util function is not specific for BDBA, right?
If so: please move to more generic module

**Review by zkdev — CHANGES_REQUESTED** [review]

File: src/components/dependencies/BDBARescoring.js (hunk 2 above)
Comment (zkdev) [reply src/components/dependencies/BDBARescoring.js:396]: Fair enough, but isn't it more intuitive to only filter "bad" sprints, rather than resetting the entire selection?

Comment (8R0WNI3) [reply src/components/dependencies/BDBARescoring.js:396]: You're probably right, my thought was that if you have a selection where some parts are "bad" (i.e. "bad" sprints defined as search param), you probably don't want this selection in place at all (as it might have been added for another scenario).

**Review by zkdev — APPROVED** [review]
/lgtm
